pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -265,6 +278,28 @@ func (c *StaticResourceController) Sync(ctx context.Context, syncContext factory
 	errors := []error{}
 	var notFoundErrorsCount int
 	for _, conditionalManifest := range c.manifests {
+		ready, err := c.precondition(ctx, conditionalManifest.files)
+		if err != nil {
+			if _, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+				Status:  operatorv1.ConditionFalse,
+				Reason:  "ErrorCheckingPrecondition",
+				Message: err.Error(),
+			})); updateErr != nil {
+				return updateErr
+			}
+			return err
+		}
+		if !ready {
+			if _, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+				Status:  operatorv1.ConditionFalse,
+				Reason:  "PreconditionNotReady",
+				Message: "PreconditionNotReady",
```

Comment:
message should be human readable

Reply:
In this case, we can not tell anything more than the precondition is not ready.

This statement is a safety net if someone doesn't follow the convention described in `StaticResourcesPreconditionFuncType` and did not return an error (with proper description):

> // StaticResourcesPreconditionFuncType checks if precondition is true and then proceeds with the reconciliation loop.
> // In case, the returned ready flag is false, a proper error with a valid description is required.

In such a case, you will fallback here and at this point, we only know that precondition was set to false with no error message.

Reply:
I added a bit more details to the status update block. Should be more readable now.